File: c-plus-plus/URI_1045.cpp
/*
	Exercise of Uri Judge Online.
	Implementation of a program that tests whether three sides form a triangle.
	By Lucas Neves.
*/

#include <iostream>
using namespace std;

void Test(double a, double b, double c) {
	if (a >= (b + c)) {
		cout << "NAO FORMA TRIANGULO" << endl;
	} else {
		if (a*a < (b*b + c*c)) {
			cout << "TRIANGULO ACUTANGULO" << endl;
		}
		if (a*a == (b*b + c*c)) {
			cout << "TRIANGULO RETANGULO" << endl;
		}
		if (a*a > (b*b + c*c)) {
			cout << "TRIANGULO OBTUSANGULO" << endl;
		}
		if ((a == b && b != c) || (a == c && b != c) || (b == c && a != b)) {
			cout << "TRIANGULO ISOSCELES" << endl;
		}
		if (a == b && b == c) {
			cout << "TRIANGULO EQUILATERO" << endl;
		}
	}
}

void Sort(double a, double b, double c) {
	if ((a >= b) && (a >= c)) {
		Test(a, b, c);
	} else if ((b > a) && (b >= c)) {
		Test(b, a, c);
	} else if ((c > a) && (c > b)) {
		Test(c, a, b);
	}
}

int main() {
	double a, b, c;
	cin >> a >> b >> c;

	Sort(a, b, c);

	return 0;
}

File: ruby/URI_1045.rb
def ordenaVetor(a, b, c)
	if (a < b)
		a, b = b, a
	end

	if (b < c)
		b, c = c, b
	end

	if (a < c)
		a, c = c, a
	end

	return a, b, c
end

def verificaTriangulo(a, b, c)
	if (a >= b + c) then
		puts "NAO FORMA TRIANGULO"
	else

		if (a**2 == b**2 + c**2) then
			puts "TRIANGULO RETANGULO"
		end

		if (a**2 > b**2 + c**2) then
			puts "TRIANGULO OBTUSANGULO"
		end

		if (a**2 < b**2 +c**2)
			puts "TRIANGULO ACUTANGULO"
		end
		
		if (a == b && b == c) then
			puts "TRIANGULO EQUILATERO"
		end

		if ((a == b && a != c) || (b == c && a != b) || (a == c && a != b))
			puts "TRIANGULO ISOSCELES"
		end
	end
end

valores = gets.split

a = valores[2].to_f
b = valores[1].to_f
c = valores[0].to_f

a, b, c = ordenaVetor(a, b, c)

verificaTriangulo(a, b, c)